addons/udes_stock/models/stock_picking.py

```diff
@@ -1342,7 +1352,22 @@ def _new_picking_for_group(self, group_key, moves, **kwargs):
             # NB: only workable pickings
             ('state', 'in', ['assigned', 'confirmed', 'waiting']),
         ])
+
+        if not picking:
+            # Otherwise reuse the existing picking if all the moves
+            # already belong to it and it contains no other moves
+            # The picking_type_id, location_id and location_dest_id
+            # will match already
+            current_picking = moves.mapped('picking_id')
+            if (len(current_picking) == 1 and
+                current_picking.mapped('move_lines') == moves):
+                current_picking.write({ 'group_id': group.id })
+                current_picking.write(kwargs)
```

Comment:
each call to `write()` here will result in a trip to the database. Better to copy/update kwargs with the new value then `write` once.